### PR TITLE
Fix randomly failing delayed_receive test in PEP suite

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR changes the `make_friends` function in `pep_SUITE` to be less prone to the race conditions that might cause the `delayed_receive` and `delayed_receive_with_sm` tests to fail. It also changes `end_per_group` function so that it always restores dynamic modules. The tests were repeated 100 times on the CI for additional certainty.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165689) / small_tests / 0cfbf7c180e2e08c4e04dc60aac0cbe1c7ac00b5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3961/165689/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3961/165689/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-14_08.44.33/lib.mongooseim.logs/run.2023-02-14_08.44.34/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165719) / small_tests / 5feaf84802bb5d08f7d9e8ceddf032a22efab656
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3961/165719/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3961/165719/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-14_11.41.06/lib.mongooseim.logs/run.2023-02-14_11.41.06/suite.log.html)

File: big_tests/tests/pep_SUITE.erl [hunk 1]
```diff
@@ -508,20 +505,34 @@ send_presence(From, Type, To) ->
...
+    % returns stanzas received finally by the inviter
+    send_presence(Bob, <<"subscribe">>, Alice),
+    escalus:assert(is_iq, escalus_client:wait_for_stanza(Bob)),
+    escalus:assert(is_presence, escalus_client:wait_for_stanza(Alice)),
+    send_presence(Alice, <<"subscribed">>, Bob),
+    escalus:assert(is_iq, escalus_client:wait_for_stanza(Alice)),
+    escalus:assert_many([is_message, is_iq]
+                        ++ lists:duplicate(2, is_presence),
+                        BobStanzas = escalus_client:wait_for_stanzas(Bob, 4)),
+    lists:filter(fun(S) -> escalus_pred:is_message(S) end, BobStanzas).
```
Comment (chrzaszcz) [inline big_tests/tests/pep_SUITE.erl:518]:
```suggestion
    lists:filter(fun escalus_pred:is_message/1, BobStanzas).
```

File: big_tests/tests/pep_SUITE.erl [hunk 2]
```diff
@@ -508,20 +505,34 @@ send_presence(From, Type, To) ->
...
-                           N =/= <<"r">>
-                 end,
-                 BobStanzas).
+    escalus:assert_many([is_iq, is_sm_ack_request],
+                        escalus_client:wait_for_stanzas(Alice, 2)),
+    escalus:assert_many([is_message, is_iq]
+                        ++ lists:duplicate(2, is_presence)
+                        ++ lists:duplicate(4, is_sm_ack_request),
+                        BobStanzas = escalus_client:wait_for_stanzas(Bob, 8)),
+    lists:filter(fun(S) -> escalus_pred:is_message(S) end, BobStanzas).
```
Comment (chrzaszcz) [inline big_tests/tests/pep_SUITE.erl:535]: As above

**Review by chrzaszcz — APPROVED** [review]
Looks good :+1:

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/165797) / small_tests / 0c23f345e7f273b158e6f9d6c60b3bdd10c1c23a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3961/165797/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3961/165797/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2023-02-15_08.25.21/lib.mongooseim.logs/run.2023-02-15_08.25.21/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]

**Pull request merged**